actionview/lib/action_view/renderer/partial_renderer.rb

```diff
@@ -521,7 +521,7 @@ def retrieve_template_keys
     def retrieve_variable(path, as)
       variable = as || begin
         base = path[-1] == "/".freeze ? "".freeze : File.basename(path)
-        raise_invalid_identifier(path) unless base =~ /\A_?(.*)(?:\.\w+)*\z/
+        raise_invalid_identifier(path) unless base =~ /\A_?(.*?)(\.\w+)*\z/
```

Comment:
Ah, sorry. I just now noticed that there’s really no need to remove `?:` from the last regex capture.